### PR TITLE
podrescli: add option to decorate a client

### PR DESCRIPTION
**Pull request by fromanirh**
This change enables the client wrapping composition at minimal cost.

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

File: pkg/resourcetopologyexporter/resourcetopologyexporter.go [hunk 1]
```diff
@@ -56,19 +58,28 @@ func ContainerIdentFromString(ident string) (*podrescli.ContainerIdent, error) {
 	return cntIdent, nil
 }
 
+func Execute(nrtupdaterArgs nrtupdater.Args, resourcemonitorArgs resourcemonitor.Args, rteArgs Args) error {
```
Comment (swatisehgal) [inline pkg/resourcetopologyexporter/resourcetopologyexporter.go:61]: We are no longer using Execute() and are now using the ExecuteWithPodresourcesClient() instead. Has this been left here intentionally or it is a mistake?

**Review by swatisehgal — COMMENTED** [review]
Looks good to me, just a minor comment.

File: pkg/resourcetopologyexporter/resourcetopologyexporter.go (hunk 1 above)
Comment (fromanirh) [reply pkg/resourcetopologyexporter/resourcetopologyexporter.go:61]: It was intentional, but on hindsight it makes no real sense. Let me rename the latter into `Execute`

**Pull request merged**